pkg/gardenlet/apis/config/v1alpha1/types.go

```diff
@@ -64,7 +64,8 @@ type GardenletConfiguration struct {
 	// +optional
 	Server *ServerConfiguration `json:"server,omitempty"`
 	// Debugging holds configuration for Debugging related features.
-	Debugging componentbaseconfigv1alpha1.DebuggingConfiguration `json:"debugging"`
+	// +optional
+	Debugging *componentbaseconfigv1alpha1.DebuggingConfiguration `json:"debugging,omitempty"`
```

Comment:
I think we have to adapt the usages of this field with checks for non-nil. Currently we have 

https://github.com/gardener/gardener/blob/238c4c04c38be9acc82e408674a0c858b07963f7/cmd/gardenlet/app/gardenlet.go#L499-L504

and when `debugging` field is not set in the component config, this will result in nil pointer dereference. The same applies also for the other components.

Reply:
Well, of course. I was too fast...